spec/operators/combineLatest-spec.ts

```diff
@@ -1,480 +1,551 @@
-    const e2subs =           '^   !';
-    const expected =         '----|';
-
-    const result = e1.pipe(combineLatest(e2, (x, y) => x + y));
-
-    expectObservable(result).toBe(expected, values);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
-    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    testScheduler.run(({ hot, expectObservable, expectSubscriptions }) => {
+      const values = {
+        a: 1,
+        b: 2,
+        c: 3,
+        r: 1 + 3 //a + c
+      };
+      const e1 = hot('-a-^-|', values);
+      const e1subs = '   ^-!';
+      const e2 = hot('-b-^-c-|', values);
+      const e2subs = '   ^---!';
+      const expected = ' ----|';
+
+      const result = e1.pipe(combineLatest(e2, (x, y) => x + y));
+
+      expectObservable(result).toBe(expected, values);
+      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+      expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    });
   });
 
   it('should work with hot-single and hot-empty', () => {
-    const values = {
-      a: 1, b: 2, c: 3
-    };
-    const e1 =        hot('-a-^-|', values);
-    const e1subs =           '^ !';
-    const e2 =        hot('-b-^-c-|', values);
-    const e2subs =           '^   !';
-    const expected =         '----|';
-
-    const result = e2.pipe(combineLatest(e1, (x, y) => x + y));
-
-    expectObservable(result).toBe(expected, values);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
-    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    testScheduler.run(({ hot, expectObservable, expectSubscriptions }) => {
+      const values = {
+        a: 1, b: 2, c: 3
+      };
+      const e1 = hot('-a-^-|', values);
+      const e1subs = '   ^-!';
+      const e2 = hot('-b-^-c-|', values);
+      const e2subs = '   ^---!';
+      const expected = ' ----|';
+
+      const result = e2.pipe(combineLatest(e1, (x, y) => x + y));
+
+      expectObservable(result).toBe(expected, values);
+      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+      expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    });
   });
 
   it('should work with hot-single and never', () => {
-    const values = {
-      a: 1
-    };
-    const e1 =        hot('-a-^-|', values);
-    const e1subs =           '^ !';
-    const e2 =        hot('------', values); //never
-    const e2subs =           '^  ';
-    const expected =         '-'; //never
-
-    const result = e1.pipe(combineLatest(e2, (x, y) => x + y));
-
-    expectObservable(result).toBe(expected, values);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
-    expectSubscriptions(e2.subscriptions).toBe(e2subs);
+    testScheduler.run(({ hot, expectObservable, expectSubscriptions }) => {
+      const values = {
+        a: 1
+      };
+      const e1 = hot('-a-^-|', values);
+      const e1subs = '   ^-!';
+      const e2 = hot('------', values); //never
+      const e2subs = '   ^--';
+      const expected = ' -'; //never
```

Comment:
As in the next test, `expected` can end where `e2` ends, but it was no impact on the test.